plugins/webapi/drng/collectivebeacon/handler.go

```diff
@@ -18,13 +19,18 @@ func Handler(c echo.Context) error {
 		return c.JSON(http.StatusBadRequest, Response{Error: err.Error()})
 	}
 
-	//TODO: to check max payload size allowed, if exceeding return an error
 	marshalUtil := marshalutil.New(request.Payload)
 	parsedPayload, err := payload.Parse(marshalUtil)
 	if err != nil {
 		return c.JSON(http.StatusBadRequest, Response{Error: "not a valid Collective Beacon payload"})
 	}
 
+	if len(parsedPayload.Bytes()) > payload.MaxCollectiveBeaconPayloadSize {
+		msg := fmt.Sprintf("maximum payload size of %d bytes exceeded", payload.MaxCollectiveBeaconPayloadSize)
+		log.Info(msg)
+		return c.JSON(http.StatusBadRequest, Response{Error: msg})
+	}
```

Comment:
The logic for whether a collective beacon payload is too big should reside within the `payload.Parse()` function.